src/stpipe/hooks.py

```diff
@@ -1,40 +1,111 @@
 """
 Pre- and post-hooks
 """
-import contextlib
-import types
+import ast
+import inspect
 
-from . import utilities
+from . import function_wrapper, utilities
 from .step import Step
 
 
-def hook_from_string(step, type, num, command):  # noqa: A002
```

Comment:
Should this be deprecated instead of removed? It looks like it's been available for a long time.

Reply:
It's not removed, I've renamed the `hook_from_string()` function to `hook_from_string_or_class()` to better describe what it actually does.  Happy to change the name back if you think this will affect the public API.  Since this is all internal workings, I would rather make all this private, including the whole `hooks.py` module, but happy to maintain backwards compat.

If not maintaining backwards compat, maybe "hook_from_object()` would be a better function name.

`stpipe` suffers from a lot of its internal workings, such as `Step.run()`, being public functions.

As far as I can tell, a search of Github doesn't turn up any uses of `stpipe.hooks` outside of `stpipe`.

Reply:
I've gone ahead and changed the function name back to its original name, and just clarified in the docstring what it does.

Reply:
Thanks! And I agree it's unlikely anyone uses it.